File: 1773/1773.py
class Solution:
    def count_items_matching_a_rule(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.count_items_matching_a_rule(root.left), self.count_items_matching_a_rule(root.right))

File: 1773/1773.go
func countItemsMatchingARule(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := countItemsMatchingARule(root.Left)
	r := countItemsMatchingARule(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}